Revit_Core_Engine/Query/DuctShape.cs

```diff
@@ -54,15 +47,28 @@ public static Autodesk.Revit.DB.ConnectorProfileType DuctShape(this Autodesk.Rev
                 return Autodesk.Revit.DB.ConnectorProfileType.Invalid;
             }
 
-            string ductFamilyName = duct?.DuctType?.FamilyName;
+            ConnectorProfileType connectorShape = ConnectorProfileType.Invalid;
 
+#if REVIT2018
+            // Get the duct connector shape in Revit 2018 by extracting it from one of the duct connectors
+            var connectors = duct.ConnectorManager.Connectors;
+            foreach (Connector connector in connectors)
+            {
+                connectorShape = connector.Shape;
+                break;
+            }
+#else
+            // Get the duct connector shape in Revit 2019 and above by extracting it from the duct type
+            connectorShape = duct.DuctType.Shape;
+#endif
+            
             // Determine the shape of the duct
-            if (ductFamilyName.Contains("Rectangular")) // Does the duct have a height
+            if (connectorShape == Autodesk.Revit.DB.ConnectorProfileType.Rectangular)
                 return Autodesk.Revit.DB.ConnectorProfileType.Rectangular; // The duct is rectangular
-            else if (ductFamilyName.Contains("Round")) // Does the duct have a diameter?
-                return Autodesk.Revit.DB.ConnectorProfileType.Round; // The duct is circular
-            else if (ductFamilyName.Contains("Oval")) // Does the duct have a diameter?
-                return Autodesk.Revit.DB.ConnectorProfileType.Oval; // The duct is circular
+            else if (connectorShape == Autodesk.Revit.DB.ConnectorProfileType.Round)
+                return Autodesk.Revit.DB.ConnectorProfileType.Round; // The duct is round
+            else if (connectorShape == Autodesk.Revit.DB.ConnectorProfileType.Oval)
+                return Autodesk.Revit.DB.ConnectorProfileType.Oval; // The duct is oval
```

Comment:
Why not `return connectorShape`?

Reply:
Good point, revised the code to return the connector shape